File: project4flutter/project4flutter/project4flutter/lib/model/Items.dart
/// item_id : 1
/// name : "hello"
/// brand : "hello"
/// description : "hello"
/// price : 2.1
/// stock : 1
/// type : "hello"
/// image : "hello"
/// is_visible : true
/// created_dt : "12/12/2004"

class Items {
  Items({
      int? itemId,
    required String? name,
    required String? brand,
    required String? description,
    required double? price,
    required int? stock,
    required String? type,
    required String? image,
    required bool? isVisible,
    DateTime? createdDt,}){

    _itemId = itemId;
    _name = name;
    _brand = brand;
    _description = description;
    _price = price;
    _stock = stock;
    _type = type;
    _image = image;
    _isVisible = isVisible;
    _createdDt = createdDt ?? DateTime.now();

}

  Items.fromJson(dynamic json) {
    _itemId = json['item_id'];
    _name = json['name'];
    _brand = json['brand'];
    _description = json['description'];
    _price = json['price'];
    _stock = json['stock'];
    _type = json['type'];
    _image = json['image'];
    _isVisible = json['is_visible'];
    _createdDt = _parseDate(json['created_dt']);
  }
  int? _itemId;
  String? _name;
  String? _brand;
  String? _description;
  double? _price;
  int? _stock;
  String? _type;
  String? _image;
  bool? _isVisible;
  DateTime? _createdDt;

  int get itemId => _itemId!;
  String get name => _name!;
  String get brand => _brand!;
  String get description => _description!;
  double get price => _price!;
  int get stock => _stock!;
  String get type => _type!;
  String get image => _image!;
  bool get isVisible => _isVisible!;
  String get createdDt => _formatDate(_createdDt)!;

  Map<String, dynamic> toJson() {
    final map = <String, dynamic>{};
    map['item_id'] = _itemId;
    map['name'] = _name;
    map['brand'] = _brand;
    map['description'] = _description;
    map['price'] = _price;
    map['stock'] = _stock;
    map['type'] = _type;
    map['image'] = _image;
    map['is_visible'] = _isVisible;
    map['created_dt'] = _formatDate(_createdDt) ?? "";
    return map;
  }
  static DateTime? _parseDate(String? dateStr) {
    if (dateStr == null || dateStr.isEmpty) return null;
    try {
      return DateTime.parse(dateStr);
    } catch (e) {
      print('Error parsing date: $e');
      return null;
    }
  }

  static String? _formatDate(DateTime? date) {
    if (date == null) return null;
    return date.toIso8601String();
  }


}

File: projectClient/src/main/java/fpt/aptech/client/models/Items.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package fpt.aptech.client.models;


import java.math.BigDecimal;

import java.util.Date;


/**
 *
 * @author Manh_Chien
 */

public class Items {
    public int item_id;
    public String name;
    public String brand;
    public String description;
    public BigDecimal price;
    public int stock;
    public int rentalquantity;
    public String type;
    public String image;
    public boolean is_visible;
    public Date created_dt;
    

    
    
    
    public Items() {
    }

    public Items(int item_id) {
        this.item_id = item_id;
        
    }

    public Items( String name, String brand, String description, BigDecimal price, int stock, int rentalquantity, String type, String image, boolean is_visible, Date created_dt) {
        
        this.name = name;
        this.brand = brand;
        this.description = description;
        this.price = price;
        this.stock = stock;
        this.rentalquantity = rentalquantity;
        this.type = type;
        this.image = image;
        this.is_visible = is_visible;
        this.created_dt = created_dt;
    }

    public Items(int item_id, String name, String brand, String description, BigDecimal price, int stock, int rentalquantity, String type, String image, boolean is_visible, Date created_dt) {
        this.item_id = item_id;
        this.name = name;
        this.brand = brand;
        this.description = description;
        this.price = price;
        this.stock = stock;
        this.rentalquantity = rentalquantity;
        this.type = type;
        this.image = image;
        this.is_visible = is_visible;
        this.created_dt = created_dt;
    }

    public int getItem_id() {
        return item_id;
    }

    public void setItem_id(int item_id) {
        this.item_id = item_id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getBrand() {
        return brand;
    }

    public void setBrand(String brand) {
        this.brand = brand;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public BigDecimal getPrice() {
        return price;
    }

    public void setPrice(BigDecimal price) {
        this.price = price;
    }

    public int getStock() {
        return stock;
    }

    public void setStock(int stock) {
        this.stock = stock;
    }

    public int getRentalquantity() {
        return rentalquantity;
    }

    public void setRentalquantity(int rentalquantity) {
        this.rentalquantity = rentalquantity;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getImage() {
        return image;
    }

    public void setImage(String image) {
        this.image = image;
    }

    public boolean isIs_visible() {
        return is_visible;
    }

    public void setIs_visible(boolean is_visible) {
        this.is_visible = is_visible;
    }

    public Date getCreated_dt() {
        return created_dt;
    }

    public void setCreated_dt(Date created_dt) {
        this.created_dt = created_dt;
    }
    

    

    

    
    
    
}
